pkg/virt-controller/watch/migration/migration.go

```diff
@@ -1060,11 +1123,50 @@ func (c *Controller) handleTargetPodCreation(key string, migration *virtv1.Virtu
 				return nil
 			}
 		}
+		err = c.handleBackendStorage(migration, vmi)
+		if err != nil {
+			return err
+		}
 		return c.createTargetPod(migration, vmi, sourcePod)
 	}
 	return nil
 }
 
+func (c *Controller) handleBackendStorage(migration *virtv1.VirtualMachineInstanceMigration, vmi *virtv1.VirtualMachineInstance) error {
+	if backendstorage.IsBackendStorageNeededForVMI(&vmi.Spec) {
+		if migration.Status.MigrationState == nil {
+			migration.Status.MigrationState = &virtv1.VirtualMachineInstanceMigrationState{}
+		}
+		migration.Status.MigrationState.SourcePersistentStatePVCName = backendstorage.CurrentPVCName(vmi)
+		if migration.Status.MigrationState.SourcePersistentStatePVCName == "" {
+			return fmt.Errorf("no backend-storage PVC found in VMI volume status")
+		}
+
+		migration.Status.MigrationState.TargetPersistentStatePVCName = backendstorage.PVCForMigrationTarget(c.pvcStore, migration)
+		if migration.Status.MigrationState.TargetPersistentStatePVCName == "" {
+			bs := backendstorage.NewBackendStorage(c.clientset, c.clusterConfig, c.storageClassStore, c.storageProfileStore, c.pvcStore)
+			vmiKey, err := controller.KeyFunc(vmi)
+			if err != nil {
+				return err
+			}
+			c.pvcExpectations.ExpectCreations(vmiKey, 1)
+			backendStoragePVC, err := bs.CreatePVCForMigrationTarget(vmi, migration.Name)
+			if err != nil {
+				c.pvcExpectations.CreationObserved(vmiKey)
+				return err
+			}
+			bs.UpdateVolumeStatus(vmi, backendStoragePVC)
+			migration.Status.MigrationState.TargetPersistentStatePVCName = backendStoragePVC.Name
+			if migration.Status.MigrationState.SourcePersistentStatePVCName == migration.Status.MigrationState.TargetPersistentStatePVCName {
```

Comment:
What do you think about 

```go
func (c *Controller) handleBackendStorage(migration *virtv1.VirtualMachineInstanceMigration, vmi *virtv1.VirtualMachineInstance) error {
	if backendstorage.IsBackendStorageNeededForVMI(&vmi.Spec) {
		if migration.Status.MigrationState == nil {
			migration.Status.MigrationState = &virtv1.VirtualMachineInstanceMigrationState{}
		}
		migration.Status.MigrationState.SourcePersistentStatePVCName = backendstorage.CurrentPVCName(vmi)
		if migration.Status.MigrationState.SourcePersistentStatePVCName == "" {
			return fmt.Errorf("no backend-storage PVC found in VMI volume status")
		}

		migration.Status.MigrationState.TargetPersistentStatePVCName = backendstorage.PVCForMigrationTarget(c.pvcStore, migration)
		if migration.Status.MigrationState.TargetPersistentStatePVCName != "" {
			// backend storage pvc has already been created or has ReadWriteMany access-mode
			return nil
		}

		bs := backendstorage.NewBackendStorage(c.clientset, c.clusterConfig, c.storageClassStore, c.storageProfileStore, c.pvcStore)
		vmiKey, err := controller.KeyFunc(vmi)
		if err != nil {
			return err
		}
		c.pvcExpectations.ExpectCreations(vmiKey, 1)
		backendStoragePVC, err := bs.CreatePVCForMigrationTarget(vmi, migration.Name)
		if err != nil {
			c.pvcExpectations.CreationObserved(vmiKey)
			return err
		}
		bs.UpdateVolumeStatus(vmi, backendStoragePVC)
		migration.Status.MigrationState.TargetPersistentStatePVCName = backendStoragePVC.Name
	}

	return nil
}
```
?

There's no need to explicitly check for the RWX case and observe the expectation.

Reply:
It's a semplification, with my suggestion you're not even creating the expectation.

Reply:
We need to call `CreatePVCForMigrationTarget` to know if target == source...

Reply:
Oh my bad you look for a different label in `PVCForMigrationTarget` ok never mind.